binary_search_tree/tree.py

```diff
@@ -8,48 +8,151 @@ def __init__(self, key, val = None):
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: On
+    # Space Complexity: O1
     def add(self, key, value = None):
```

Comment:
👍 However space complexity is O(log n) if the tree is balanced and O(n) otherwise due to the recursive call stack.